src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Window.cs

```diff
@@ -353,12 +353,12 @@ public Nullable<bool> ShowDialog()
             Debug.Assert(_threadWindowHandles == null, "_threadWindowHandles must be null before enumerating the thread windows");
 
             // NOTE:
-            // _threadWindowHandles is created here.  This reference is nulled out in EnableThreadWindows
-            // when it is called with a true parameter.  Please do not null it out anywhere else.
-            // EnableThreadWindow(true) is called when dialog is going away.  Once dialog is closed and
-            // thread windows have been enabled, then there no need to keep the array list around.
+            // _threadWindowHandles is created here. This reference is nulled out in EnableThreadWindows
+            // when it is called with a true parameter. Please do not null it out anywhere else.
+            // EnableThreadWindow(true) is called when dialog is going away. Once dialog is closed and
+            // thread windows have been enabled, then there no need to keep the list around.
             // Please see BUG 929740 before making any changes to how _threadWindowHandles works.
-            _threadWindowHandles = new ArrayList();
+            _threadWindowHandles = new();
```

Comment:
Nit: in latest C#, using `_threadWindowHandles = []` can be more natural.

Reply:
I've decided to go with the least confusing in the end.